### ISSUE TITLE
RagdollUtils can't handle 16-bit bone indices

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#1004]
Models imported using the `GltfLoader` typically have 16-bit bone indices.
If a `KinematicRagdollControl` is added to such a model, a `ClassCastException` is thrown in `RagdollUtils`.

Here is a typical stack trace, courtesy of oxplay2:
```
SEVERE: Uncaught exception thrown in Thread[jME3 Main,5,main]
java.lang.ClassCastException: java.nio.DirectShortBufferU cannot be cast to java.nio.ByteBuffer
at com.jme3.bullet.control.ragdoll.RagdollUtils.getPoints(RagdollUtils.java:229)
at com.jme3.bullet.control.ragdoll.RagdollUtils.makeShapeFromVerticeWeights(RagdollUtils.java:203)
at com.jme3.bullet.control.KinematicRagdollControl.boneRecursion(KinematicRagdollControl.java:534)
at com.jme3.bullet.control.KinematicRagdollControl.scanSpatial(KinematicRagdollControl.java:515)
at com.jme3.bullet.control.KinematicRagdollControl.createSpatialData(KinematicRagdollControl.java:467)
at com.jme3.bullet.control.AbstractPhysicsControl.setSpatial(AbstractPhysicsControl.java:182)
at com.jme3.scene.Spatial.addControl(Spatial.java:769)
at test.RagdollTest3.simpleInitApp(RagdollTest3.java:130)
at com.jme3.app.SimpleApplication.initialize(SimpleApplication.java:220)
at com.jme3.system.lwjgl.LwjglAbstractDisplay.initInThread(LwjglAbstractDisplay.java:130)
at com.jme3.system.lwjgl.LwjglAbstractDisplay.run(LwjglAbstractDisplay.java:211)
at java.lang.Thread.run(Thread.java:748)
```

**Comment by oxplay2** [comment]
it was simple cube with one bone(p -> automatic weight) using Blender 2.79b.

Was working using blender exporter, but using gltf exporter was above exception.

Anyway KinematicRagdollControl will be replaced by DynamicAnimControl?
based on "If you’re on JME 3.2, you can start using DynamicAnimControl any time you want. You don’t have to wait for JME 3.3 to be released."

but i agree, it might take long time before 3.3

**Comment by stephengold** [comment]
Thanks for the additional info. I've created my own test case.

This issue should be fixed in 3.2.3, if that ever happens.

**Comment by stephengold** [comment]
Fixed in `master` branch at hash 51b12e1b9e4a5a22708db8b36500d854fc76f7d9.
